craft_parts/executor/part_handler.py

```diff
@@ -76,6 +76,25 @@ def __init__(
         self._build_packages = _get_build_packages(part=self._part, plugin=self._plugin)
         self._build_snaps = _get_build_snaps(part=self._part, plugin=self._plugin)
 
+    @property
+    def build_packages(self) -> List[str]:
+        """Return the list of build packages defined for this part.
+
+        The list of build packages include packages defined directly in
+        the parts specification, packages required by the source handler,
+        and packages required by the plugin.
+        """
+        return self._build_packages
```

Comment:
What's the point of these properties? Can't we just make `self._build_packages` public?

Note that you're not protecting this attribute of being modified, as it's mutable, so there's no point in protecting it from somebody doing `part_handler.build_packages = []` when they still could do `part_handler.build_packages.clear()`.

Reply:
Good point. Removing them.